ASP.NET API/WAVC_WebApi/Controllers/AuthenticationControllers/LoginController.cs

```diff
@@ -0,0 +1,69 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IdentityModel.Tokens.Jwt;
+using System.Linq;
+using System.Security.Claims;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Options;
+using Microsoft.IdentityModel.Tokens;
+using WAVC_WebApi.Data;
+using WAVC_WebApi.Models;
+
+namespace WAVC_WebApi.Controllers.AuthenticationControllers
+{
+    [Route("api/[controller]")]
+    [ApiController]
+    public class LoginController : ControllerBase
+    {
+        private readonly UserManager<ApplicationUser> _userManager;
+        private readonly ApplicationSettings _applicationSettings;
+
+        public LoginController(
+            UserManager<ApplicationUser> userManager, IOptions<ApplicationSettings> appSettings)
+        {
+            _userManager = userManager;
+            _applicationSettings = appSettings.Value;
+        }
+
+        [HttpPost]
+        //POST : /api/ApplicationUser/Login
+        public async Task<IActionResult> Login(LoginModel model)
+        {
+            ApplicationUser user = await _userManager.FindByEmailAsync(model.Email);
+
+            if (user == null || await _userManager.CheckPasswordAsync(user, model.Password) == false)
+                return BadRequest(new { message = "Username or password is incorrect." });
+
+            string token = GenerateToken(user);
+
+            return Ok(new { token });
```

Comment:
I told you it's gonna be done in a separate PR.

Reply:
It's not safer btw. Cookies and JWT in a local storage are just vulnerable to different attacks. There is no perfect solution. Either you're team cookie or team JWT.

Reply:
With cookies you can just set some flags (that asp.net core probably sets for us) and it's done. Right now your inteceptor tries to imitate cookies, but it doesn't have such flags.

Reply:
It simply doesn't need such flags because it's not vulnerable to these kinds of attack. Guards, interceptors and jwt token in a local storage aren't just my idea. This solution is recommended by microsoft, and angular themself when it comes to jwt authorization. I'm sick of this discussion, over and over again. I told you it's going to be done via cookies so it's gonna be done. But now I have some other priorities, like provide basic functionalities, rather than messing around with an  authorization for an app which is being developed for academic purposes only. Calm down and let us do our work. Authorization can be changed afterwards and any potential refactor (it shouldn't affect anything else than this one line in each controller) I take on me. I assure you it's going to be done your way but firstly I need to focus on chat and other things. Peace.

Reply:
"It simply doesn't need such flags because it's not vulnerable to these kinds of attack." - that's untrue. Tokens shouldn't be intercepted by hacker, because that will allow him to impersonate user.
1. sending them over http is unsecure -  https should be ensured,
2. possibility to send them to other servers is unsecure - interceptor should check where token is going to be sent,
3. tokens are aviable on JavaScript level, any code injection can retrive them.
Your solution doesn't provide anything to prevent those vulnerabilities, while cookie flags do. I don't know where you got information that microsoft and angular recommend that, but I hope you're not interpreting something wrongly. I know that solution isn't just your idea, but often for purposes of simple tutorial things are simplified or fact, that other people do it doesn't mean it's the best way.
"Rather than messing around with an authorization for an app which is being developed for academic purposes only. Calm down and let us do our work" - may I kindly remind you that I was supposed to do authorization.

Reply:
PS. The only drawback I've found was in Microsofts documentation is that if non-web app would need talk ONLY with SignalR (which is definitely NOT the case here), API to exchange authentication data would have to be implemented.

Reply:
PS2. "I told you it's going to be done via cookies so it's gonna be done" I'm glad that you agreed to use cookies, but I also want you to understand that I'm not arguing just, because I want things my way, but because I genuinely believe (and not that my beliefs are 100% true, I just wasn't convinced to be otherwise) that this solution is better.

Reply:
I perfectly know that this solution is vulnerable to XSS attack but it is resistant to other attacks that cookies are not. Nonetheless I don't want to continue this discussion anymore. Authorization is yours. Do whatever you want.